src/awesome/console/AwesomeLinkFilter.java

```diff
@@ -162,6 +166,25 @@ public String getFileFromUrl(final String url) {
 		return results;
 	}
 
+	private List<VirtualFile> strictFilesByPaths(final List<VirtualFile> matchingFiles, final FileLinkMatch match) {
+		List<VirtualFile> strictMatchingFiles = new ArrayList<>();
+		for (VirtualFile matchedFile : matchingFiles) {
+			String generalisedFilePath = generalisePath(matchedFile.getPath());
+			String generalisedMatchPath = generalisePath(match.path);
+			if (generalisedFilePath.endsWith(generalisedMatchPath)) {
+				strictMatchingFiles.add(matchedFile);
+			}
+		}
+		return strictMatchingFiles;
+	}
+
+	private String generalisePath(final String path) {
+		return path.replace('/', '.')
```

Comment:
What exactly are you doing here, why shall / and \ be replaced by a dot? and why is ~ a , (comma)?
What happens if my files is named `..file`

Reply:
To avoid some differences in folder seperators ('/', '\') I was simply replace them as '.'
```
.replace('~', ',')
.replace("..","");
```
was introduced in separate commit to resolve @twidi issue.
Problem were that file `~/test.py` wasn't same as `/home/jme/test.py`, so my little dirty trick was to kinda `trim` away path aliases by replacing them as `.` and remove them.

There would be issue if You have two files: `..file` and `file`, than it would open popup with both of them. But If You would have only one of them than it would be ok, because it would be `trimmed` from both: `match` and `VirtualFile matchedFile`, so they would be equal.

Reply:
But I'll change it a little bit. I gonna try to resolve `FileLinkMatch match` full path and if it can be resolved and file exists, than I could use it to compare. That would resolve path aliase `~`. It would be tricky with path alias `.` because it's relative to current path.

Reply:
Maybe a solution is doing it in a iterative way.

if clicked path is `../foo/bar/file.py` and the "know" files are `foo/bar/file.py` and `foo/qux/file.py`:

start with `file.py` => i have two matching files, go back one dir => check with `bar/file.py` => ok I have only one matching file, we're good

just an idea, I don't know 1/ if it's really a good idea, and 2/ if it's feasible

Reply:
Sorry, I responded in new comment. But I gonna try out this idea in following weekends (hopefuly) :)

Reply:
@Rublis thanks for your effort, much appreciated. Waiting until you notify me :cat: would be nice if the matching and trimming would then also work without miss-detecting `..test.py` and `test.py` simultaneously.